### ISSUE TITLE
Require implementation of syscalls 213, 291 (epoll_create & epoll_create1)

### ISSUE THREAD

**Issue by MikeCamel** [enarx/enarx-keepldr#259]
Required to create a listening socket in a Keep.

```
Dec 15 16:40:55 raption enarx-keepldr[142419]: init_gdt
Dec 15 16:40:55 raption enarx-keepldr[142419]: mmap(0x0, 0x890, 0x3, 0x22, 0xffffffffffffffff, 0x0)
Dec 15 16:40:55 raption enarx-keepldr[142419]: SC> mmap(0, 2192, …) = 0x00007f2f2c0ae000
Dec 15 16:40:55 raption enarx-keepldr[142419]: arch_prctl(0x1002, 0x7f2f2c0ae7a0)
Dec 15 16:40:55 raption enarx-keepldr[142419]: SC> arch_prctl(ARCH_SET_FS, 0x7f2f2c0ae7a0) = 0
Dec 15 16:40:55 raption enarx-keepldr[142419]: set_tid_address(0x7f2f2c0ad0a0)
Dec 15 16:40:55 raption enarx-keepldr[142419]: poll(0x7ff0b7177d00, 0x3, 0x0)
Dec 15 16:40:55 raption enarx-keepldr[142419]: rt_sigaction(0xd, 0x7ff0b7177b40, 0x7ff0b7177b60, 0x8)
Dec 15 16:40:55 raption enarx-keepldr[142419]: rt_sigaction(0xb, 0x0, 0x7ff0b7177ca0, 0x8)
Dec 15 16:40:55 raption enarx-keepldr[142419]: rt_sigprocmask(0x1, 0x7ff0b7177ca0, 0x0, 0x8)
Dec 15 16:40:55 raption enarx-keepldr[142419]: rt_sigaction(0xb, 0x7ff0b7177c80, 0x0, 0x8)
Dec 15 16:40:55 raption enarx-keepldr[142419]: rt_sigaction(0x7, 0x0, 0x7ff0b7177ca0, 0x8)
Dec 15 16:40:55 raption enarx-keepldr[142419]: rt_sigaction(0x7, 0x7ff0b7177c80, 0x0, 0x8)
Dec 15 16:40:55 raption enarx-keepldr[142419]: sigaltstack(0x0, 0x7ff0b7177cc0)
Dec 15 16:40:55 raption enarx-keepldr[142419]: brk(0x0)
Dec 15 16:40:55 raption enarx-keepldr[142419]: SC> brk(0x0000000000000000) = 0x555522f4f000
Dec 15 16:40:55 raption enarx-keepldr[142419]: brk(0x555522f50000)
Dec 15 16:40:55 raption enarx-keepldr[142419]: SC> brk(0x0000555522f50000) = 0x555522f50000
Dec 15 16:40:55 raption enarx-keepldr[142419]: rt_sigprocmask(0x0, 0x7f2f2bbafac0, 0x7ff0b7177b00, 0x8)
Dec 15 16:40:55 raption enarx-keepldr[142419]: rt_sigprocmask(0x2, 0x7ff0b7177b00, 0x0, 0x8)
Dec 15 16:40:55 raption enarx-keepldr[142419]: unsupported syscall: 2
Dec 15 16:40:55 raption enarx-keepldr[142419]: unsupported syscall: 204
Dec 15 16:40:55 raption enarx-keepldr[142419]: unsupported syscall: 204
Dec 15 16:40:55 raption enarx-keepldr[142419]: unsupported syscall: 291
Dec 15 16:40:55 raption enarx-keepldr[142419]: unsupported syscall: 213
```

**Comment by haraldh** [comment]
what is this? wasm?

**Comment by MikeCamel** [comment]
This fails when running in a KVM keep, but is fine in a Nil Keep: https://github.com/MikeCamel/enarx-wasmldr/blob/https-wasmldr-exp/src/main.rs

**Comment by haraldh** [comment]
Why don't you use `0.0.0.0` as the listen address??

https://github.com/MikeCamel/enarx-wasmldr/blob/caece8b3cc7817749edffcef337c9f2d431d9718/src/main.rs#L77

**Comment by MikeCamel** [comment]
Because one of the tests I've been running has been explicitly testing a client on a separate machine (and I've found bugs that way). We will need to deecide how the keepldr decides which address to bind on (probably via a NATted bridge), but for now, this is easiest. The other components I'm working on have config files to specify this, but it's not appropriate for this one.

**Comment by haraldh** [comment]
> Because one of the tests I've been running has been explicitly testing a client on a separate machine (and I've found bugs that way). We will need to deecide how the keepldr decides which address to bind on (probably via a NATted bridge), but for now, this is easiest. The other components I'm working on have config files to specify this, but it's not appropriate for this one.

But `0.0.0.0` listens on every IP address... How does this not work?

**Comment by MikeCamel** [comment]
I want to restrict where I'm listening, and also to produce certificates to tie to a particular hostname/IP address combo.  Point taken, though.

**Comment by haraldh** [comment]
I don't know... the outside world could map internal IP addresses to external ones as it wants. Think round-robin. I don't think our service should know the exact IP address or hostname.

**Comment by haraldh** [comment]
So, I implemented `pipe` and `epoll_*` ... The next thing tokio wants is threading... which is a rather big thing to implement.

**Comment by MikeCamel** [comment]
Are these being added separately? Do we have a status, or should they have been merged with #261?

**Comment by haraldh** [comment]
It's in #263
